// @paypal/paypal-server-sdk/dist/esm/models/capturedPayment.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema.js';
import { captureStatusSchema } from './captureStatus.js';
import { captureStatusDetailsSchema, } from './captureStatusDetails.js';
import { disbursementModeSchema, } from './disbursementMode.js';
import { linkDescriptionSchema } from './linkDescription.js';
import { moneySchema } from './money.js';
import { networkTransactionSchema, } from './networkTransaction.js';
import { payeeBaseSchema } from './payeeBase.js';
import { paymentSupplementaryDataSchema, } from './paymentSupplementaryData.js';
import { processorResponseSchema, } from './processorResponse.js';
import { sellerProtectionSchema, } from './sellerProtection.js';
import { sellerReceivableBreakdownSchema, } from './sellerReceivableBreakdown.js';
export const capturedPaymentSchema = lazy(() => object({
    status: ['status', optional(captureStatusSchema)],
    statusDetails: ['status_details', optional(captureStatusDetailsSchema)],
    id: ['id', optional(string())],
    amount: ['amount', optional(moneySchema)],
    invoiceId: ['invoice_id', optional(string())],
    customId: ['custom_id', optional(string())],
    networkTransactionReference: [
        'network_transaction_reference',
        optional(networkTransactionSchema),
    ],
    sellerProtection: ['seller_protection', optional(sellerProtectionSchema)],
    finalCapture: ['final_capture', optional(boolean())],
    sellerReceivableBreakdown: [
        'seller_receivable_breakdown',
        optional(sellerReceivableBreakdownSchema),
    ],
    disbursementMode: ['disbursement_mode', optional(disbursementModeSchema)],
    links: ['links', optional(array(linkDescriptionSchema))],
    processorResponse: [
        'processor_response',
        optional(processorResponseSchema),
    ],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    supplementaryData: [
        'supplementary_data',
        optional(paymentSupplementaryDataSchema),
    ],
    payee: ['payee', optional(payeeBaseSchema)],
}));
