// @paypal/paypal-server-sdk/dist/esm/models/cardResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { authenticationResponseSchema, } from './authenticationResponse.js';
import { binDetailsSchema } from './binDetails.js';
import { cardAttributesResponseSchema, } from './cardAttributesResponse.js';
import { cardBrandSchema } from './cardBrand.js';
import { cardFromRequestSchema } from './cardFromRequest.js';
import { cardStoredCredentialSchema, } from './cardStoredCredential.js';
import { cardTypeSchema } from './cardType.js';
export const cardResponseSchema = lazy(() => object({
    name: ['name', optional(string())],
    lastDigits: ['last_digits', optional(string())],
    brand: ['brand', optional(cardBrandSchema)],
    availableNetworks: ['available_networks', optional(array(cardBrandSchema))],
    type: ['type', optional(cardTypeSchema)],
    authenticationResult: [
        'authentication_result',
        optional(authenticationResponseSchema),
    ],
    attributes: ['attributes', optional(cardAttributesResponseSchema)],
    fromRequest: ['from_request', optional(cardFromRequestSchema)],
    expiry: ['expiry', optional(string())],
    binDetails: ['bin_details', optional(binDetailsSchema)],
    storedCredential: [
        'stored_credential',
        optional(cardStoredCredentialSchema),
    ],
}));
