// @paypal/paypal-server-sdk/dist/esm/models/createSubscriptionRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, lazy, object, optional, string } from '../schema.js';
import { moneySchema } from './money.js';
import { planOverrideSchema } from './planOverride.js';
import { subscriberRequestSchema, } from './subscriberRequest.js';
import { subscriptionApplicationContextSchema, } from './subscriptionApplicationContext.js';
export const createSubscriptionRequestSchema = lazy(() => object({
    planId: ['plan_id', string()],
    startTime: ['start_time', optional(string())],
    quantity: ['quantity', optional(string())],
    shippingAmount: ['shipping_amount', optional(moneySchema)],
    subscriber: ['subscriber', optional(subscriberRequestSchema)],
    autoRenewal: ['auto_renewal', optional(boolean())],
    applicationContext: [
        'application_context',
        optional(subscriptionApplicationContextSchema),
    ],
    customId: ['custom_id', optional(string())],
    plan: ['plan', optional(planOverrideSchema)],
}));
