// @paypal/paypal-server-sdk/dist/esm/models/googlePayRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { assuranceDetailsSchema, } from './assuranceDetails.js';
import { googlePayDecryptedTokenDataSchema, } from './googlePayDecryptedTokenData.js';
import { googlePayExperienceContextSchema, } from './googlePayExperienceContext.js';
import { googlePayRequestCardSchema, } from './googlePayRequestCard.js';
import { phoneNumberWithCountryCodeSchema, } from './phoneNumberWithCountryCode.js';
export const googlePayRequestSchema = lazy(() => object({
    name: ['name', optional(string())],
    emailAddress: ['email_address', optional(string())],
    phoneNumber: ['phone_number', optional(phoneNumberWithCountryCodeSchema)],
    card: ['card', optional(googlePayRequestCardSchema)],
    decryptedToken: [
        'decrypted_token',
        optional(googlePayDecryptedTokenDataSchema),
    ],
    assuranceDetails: ['assurance_details', optional(assuranceDetailsSchema)],
    experienceContext: [
        'experience_context',
        optional(googlePayExperienceContextSchema),
    ],
}));
