// @paypal/paypal-server-sdk/dist/esm/models/itemDetails.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { checkoutOptionSchema } from './checkoutOption.js';
import { moneySchema } from './money.js';
import { taxAmountSchema } from './taxAmount.js';
export const itemDetailsSchema = lazy(() => object({
    itemCode: ['item_code', optional(string())],
    itemName: ['item_name', optional(string())],
    itemDescription: ['item_description', optional(string())],
    itemOptions: ['item_options', optional(string())],
    itemQuantity: ['item_quantity', optional(string())],
    itemUnitPrice: ['item_unit_price', optional(moneySchema)],
    itemAmount: ['item_amount', optional(moneySchema)],
    discountAmount: ['discount_amount', optional(moneySchema)],
    adjustmentAmount: ['adjustment_amount', optional(moneySchema)],
    giftWrapAmount: ['gift_wrap_amount', optional(moneySchema)],
    taxPercentage: ['tax_percentage', optional(string())],
    taxAmounts: ['tax_amounts', optional(array(taxAmountSchema))],
    basicShippingAmount: ['basic_shipping_amount', optional(moneySchema)],
    extraShippingAmount: ['extra_shipping_amount', optional(moneySchema)],
    handlingAmount: ['handling_amount', optional(moneySchema)],
    insuranceAmount: ['insurance_amount', optional(moneySchema)],
    totalItemAmount: ['total_item_amount', optional(moneySchema)],
    invoiceNumber: ['invoice_number', optional(string())],
    checkoutOptions: [
        'checkout_options',
        optional(array(checkoutOptionSchema)),
    ],
}));
