// @paypal/paypal-server-sdk/dist/esm/models/lineItem.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { moneySchema } from './money.js';
import { orderBillingPlanSchema, } from './orderBillingPlan.js';
import { universalProductCodeSchema, } from './universalProductCode.js';
export const lineItemSchema = lazy(() => object({
    name: ['name', string()],
    quantity: ['quantity', string()],
    description: ['description', optional(string())],
    sku: ['sku', optional(string())],
    url: ['url', optional(string())],
    imageUrl: ['image_url', optional(string())],
    upc: ['upc', optional(universalProductCodeSchema)],
    billingPlan: ['billing_plan', optional(orderBillingPlanSchema)],
    unitAmount: ['unit_amount', optional(moneySchema)],
    tax: ['tax', optional(moneySchema)],
    commodityCode: ['commodity_code', optional(string())],
    discountAmount: ['discount_amount', optional(moneySchema)],
    totalAmount: ['total_amount', optional(moneySchema)],
    unitOfMeasure: ['unit_of_measure', optional(string())],
}));
