// @paypal/paypal-server-sdk/dist/esm/models/paymentSourceResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional } from '../schema.js';
import { applePayPaymentObjectSchema, } from './applePayPaymentObject.js';
import { bancontactPaymentObjectSchema, } from './bancontactPaymentObject.js';
import { blikPaymentObjectSchema, } from './blikPaymentObject.js';
import { cardResponseSchema } from './cardResponse.js';
import { epsPaymentObjectSchema, } from './epsPaymentObject.js';
import { giropayPaymentObjectSchema, } from './giropayPaymentObject.js';
import { googlePayWalletResponseSchema, } from './googlePayWalletResponse.js';
import { idealPaymentObjectSchema, } from './idealPaymentObject.js';
import { mybankPaymentObjectSchema, } from './mybankPaymentObject.js';
import { p24PaymentObjectSchema, } from './p24PaymentObject.js';
import { paypalWalletResponseSchema, } from './paypalWalletResponse.js';
import { sofortPaymentObjectSchema, } from './sofortPaymentObject.js';
import { trustlyPaymentObjectSchema, } from './trustlyPaymentObject.js';
import { venmoWalletResponseSchema, } from './venmoWalletResponse.js';
export const paymentSourceResponseSchema = lazy(() => object({
    card: ['card', optional(cardResponseSchema)],
    paypal: ['paypal', optional(paypalWalletResponseSchema)],
    bancontact: ['bancontact', optional(bancontactPaymentObjectSchema)],
    blik: ['blik', optional(blikPaymentObjectSchema)],
    eps: ['eps', optional(epsPaymentObjectSchema)],
    giropay: ['giropay', optional(giropayPaymentObjectSchema)],
    ideal: ['ideal', optional(idealPaymentObjectSchema)],
    mybank: ['mybank', optional(mybankPaymentObjectSchema)],
    p24: ['p24', optional(p24PaymentObjectSchema)],
    sofort: ['sofort', optional(sofortPaymentObjectSchema)],
    trustly: ['trustly', optional(trustlyPaymentObjectSchema)],
    applePay: ['apple_pay', optional(applePayPaymentObjectSchema)],
    googlePay: ['google_pay', optional(googlePayWalletResponseSchema)],
    venmo: ['venmo', optional(venmoWalletResponseSchema)],
}));
