// @paypal/paypal-server-sdk/dist/esm/models/paymentSource.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional } from '../schema.js';
import { applePayRequestSchema } from './applePayRequest.js';
import { bancontactPaymentRequestSchema, } from './bancontactPaymentRequest.js';
import { blikPaymentRequestSchema, } from './blikPaymentRequest.js';
import { cardRequestSchema } from './cardRequest.js';
import { epsPaymentRequestSchema, } from './epsPaymentRequest.js';
import { giropayPaymentRequestSchema, } from './giropayPaymentRequest.js';
import { googlePayRequestSchema, } from './googlePayRequest.js';
import { idealPaymentRequestSchema, } from './idealPaymentRequest.js';
import { mybankPaymentRequestSchema, } from './mybankPaymentRequest.js';
import { p24PaymentRequestSchema, } from './p24PaymentRequest.js';
import { paypalWalletSchema } from './paypalWallet.js';
import { sofortPaymentRequestSchema, } from './sofortPaymentRequest.js';
import { tokenSchema } from './token.js';
import { trustlyPaymentRequestSchema, } from './trustlyPaymentRequest.js';
import { venmoWalletRequestSchema, } from './venmoWalletRequest.js';
export const paymentSourceSchema = lazy(() => object({
    card: ['card', optional(cardRequestSchema)],
    token: ['token', optional(tokenSchema)],
    paypal: ['paypal', optional(paypalWalletSchema)],
    bancontact: ['bancontact', optional(bancontactPaymentRequestSchema)],
    blik: ['blik', optional(blikPaymentRequestSchema)],
    eps: ['eps', optional(epsPaymentRequestSchema)],
    giropay: ['giropay', optional(giropayPaymentRequestSchema)],
    ideal: ['ideal', optional(idealPaymentRequestSchema)],
    mybank: ['mybank', optional(mybankPaymentRequestSchema)],
    p24: ['p24', optional(p24PaymentRequestSchema)],
    sofort: ['sofort', optional(sofortPaymentRequestSchema)],
    trustly: ['trustly', optional(trustlyPaymentRequestSchema)],
    applePay: ['apple_pay', optional(applePayRequestSchema)],
    googlePay: ['google_pay', optional(googlePayRequestSchema)],
    venmo: ['venmo', optional(venmoWalletRequestSchema)],
}));
