// @paypal/paypal-server-sdk/dist/esm/models/paypalWallet.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { nameSchema } from './name.js';
import { paypalWalletAttributesSchema, } from './paypalWalletAttributes.js';
import { paypalWalletExperienceContextSchema, } from './paypalWalletExperienceContext.js';
import { paypalWalletStoredCredentialSchema, } from './paypalWalletStoredCredential.js';
import { phoneWithTypeSchema } from './phoneWithType.js';
import { taxInfoSchema } from './taxInfo.js';
export const paypalWalletSchema = lazy(() => object({
    vaultId: ['vault_id', optional(string())],
    emailAddress: ['email_address', optional(string())],
    name: ['name', optional(nameSchema)],
    phone: ['phone', optional(phoneWithTypeSchema)],
    birthDate: ['birth_date', optional(string())],
    taxInfo: ['tax_info', optional(taxInfoSchema)],
    address: ['address', optional(addressSchema)],
    attributes: ['attributes', optional(paypalWalletAttributesSchema)],
    experienceContext: [
        'experience_context',
        optional(paypalWalletExperienceContextSchema),
    ],
    billingAgreementId: ['billing_agreement_id', optional(string())],
    storedCredential: [
        'stored_credential',
        optional(paypalWalletStoredCredentialSchema),
    ],
}));
