// @paypal/paypal-server-sdk/dist/esm/models/paypalWalletResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { experienceStatusSchema, } from './experienceStatus.js';
import { nameSchema } from './name.js';
import { paypalWalletAccountVerificationStatusSchema, } from './paypalWalletAccountVerificationStatus.js';
import { paypalWalletAttributesResponseSchema, } from './paypalWalletAttributesResponse.js';
import { paypalWalletStoredCredentialSchema, } from './paypalWalletStoredCredential.js';
import { phoneNumberSchema } from './phoneNumber.js';
import { phoneTypeSchema } from './phoneType.js';
import { taxInfoSchema } from './taxInfo.js';
export const paypalWalletResponseSchema = lazy(() => object({
    emailAddress: ['email_address', optional(string())],
    accountId: ['account_id', optional(string())],
    accountStatus: [
        'account_status',
        optional(paypalWalletAccountVerificationStatusSchema),
    ],
    name: ['name', optional(nameSchema)],
    phoneType: ['phone_type', optional(phoneTypeSchema)],
    phoneNumber: ['phone_number', optional(phoneNumberSchema)],
    birthDate: ['birth_date', optional(string())],
    businessName: ['business_name', optional(string())],
    taxInfo: ['tax_info', optional(taxInfoSchema)],
    address: ['address', optional(addressSchema)],
    attributes: [
        'attributes',
        optional(paypalWalletAttributesResponseSchema),
    ],
    storedCredential: [
        'stored_credential',
        optional(paypalWalletStoredCredentialSchema),
    ],
    experienceStatus: ['experience_status', optional(experienceStatusSchema)],
}));
