// @paypal/paypal-server-sdk/dist/esm/models/paypalWalletVaultBase.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, object, optional, string } from '../schema.js';
import { paypalPaymentTokenCustomerTypeSchema, } from './paypalPaymentTokenCustomerType.js';
import { storeInVaultInstructionSchema, } from './storeInVaultInstruction.js';
import { usagePatternSchema } from './usagePattern.js';
import { usageTypeSchema } from './usageType.js';
export const paypalWalletVaultBaseSchema = object({
    storeInVault: ['store_in_vault', optional(storeInVaultInstructionSchema)],
    description: ['description', optional(string())],
    usagePattern: ['usage_pattern', optional(usagePatternSchema)],
    usageType: ['usage_type', optional(usageTypeSchema)],
    customerType: [
        'customer_type',
        optional(paypalPaymentTokenCustomerTypeSchema),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        optional(boolean()),
    ],
});
