// @paypal/paypal-server-sdk/dist/esm/models/planDetails.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema.js';
import { merchantPreferencesSchema, } from './merchantPreferences.js';
import { paymentPreferencesSchema, } from './paymentPreferences.js';
import { subscriptionBillingCycleSchema, } from './subscriptionBillingCycle.js';
import { taxesSchema } from './taxes.js';
export const planDetailsSchema = lazy(() => object({
    productId: ['product_id', optional(string())],
    name: ['name', optional(string())],
    description: ['description', optional(string())],
    billingCycles: [
        'billing_cycles',
        optional(array(subscriptionBillingCycleSchema)),
    ],
    paymentPreferences: [
        'payment_preferences',
        optional(paymentPreferencesSchema),
    ],
    merchantPreferences: [
        'merchant_preferences',
        optional(merchantPreferencesSchema),
    ],
    taxes: ['taxes', optional(taxesSchema)],
    quantitySupported: ['quantity_supported', optional(boolean())],
}));
