// @paypal/paypal-server-sdk/dist/esm/models/purchaseUnitRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { amountWithBreakdownSchema, } from './amountWithBreakdown.js';
import { itemRequestSchema } from './itemRequest.js';
import { payeeBaseSchema } from './payeeBase.js';
import { paymentInstructionSchema, } from './paymentInstruction.js';
import { shippingDetailsSchema } from './shippingDetails.js';
import { supplementaryDataSchema, } from './supplementaryData.js';
export const purchaseUnitRequestSchema = lazy(() => object({
    referenceId: ['reference_id', optional(string())],
    amount: ['amount', amountWithBreakdownSchema],
    payee: ['payee', optional(payeeBaseSchema)],
    paymentInstruction: [
        'payment_instruction',
        optional(paymentInstructionSchema),
    ],
    description: ['description', optional(string())],
    customId: ['custom_id', optional(string())],
    invoiceId: ['invoice_id', optional(string())],
    softDescriptor: ['soft_descriptor', optional(string())],
    items: ['items', optional(array(itemRequestSchema))],
    shipping: ['shipping', optional(shippingDetailsSchema)],
    supplementaryData: [
        'supplementary_data',
        optional(supplementaryDataSchema),
    ],
}));
