// @paypal/paypal-server-sdk/dist/esm/models/sellerPayableBreakdown.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional } from '../schema.js';
import { moneySchema } from './money.js';
import { netAmountBreakdownItemSchema, } from './netAmountBreakdownItem.js';
import { platformFeeSchema } from './platformFee.js';
export const sellerPayableBreakdownSchema = lazy(() => object({
    grossAmount: ['gross_amount', optional(moneySchema)],
    paypalFee: ['paypal_fee', optional(moneySchema)],
    paypalFeeInReceivableCurrency: [
        'paypal_fee_in_receivable_currency',
        optional(moneySchema),
    ],
    netAmount: ['net_amount', optional(moneySchema)],
    netAmountInReceivableCurrency: [
        'net_amount_in_receivable_currency',
        optional(moneySchema),
    ],
    platformFees: ['platform_fees', optional(array(platformFeeSchema))],
    netAmountBreakdown: [
        'net_amount_breakdown',
        optional(array(netAmountBreakdownItemSchema)),
    ],
    totalRefundedAmount: ['total_refunded_amount', optional(moneySchema)],
}));
