// @paypal/paypal-server-sdk/dist/esm/models/setupTokenRequestPaymentSource.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional } from '../schema.js';
import { bankRequestSchema } from './bankRequest.js';
import { setupTokenRequestCardSchema, } from './setupTokenRequestCard.js';
import { vaultApplePayRequestSchema, } from './vaultApplePayRequest.js';
import { vaultPaypalWalletRequestSchema, } from './vaultPaypalWalletRequest.js';
import { vaultTokenRequestSchema, } from './vaultTokenRequest.js';
import { vaultVenmoRequestSchema, } from './vaultVenmoRequest.js';
export const setupTokenRequestPaymentSourceSchema = lazy(() => object({
    card: ['card', optional(setupTokenRequestCardSchema)],
    paypal: ['paypal', optional(vaultPaypalWalletRequestSchema)],
    venmo: ['venmo', optional(vaultVenmoRequestSchema)],
    applePay: ['apple_pay', optional(vaultApplePayRequestSchema)],
    token: ['token', optional(vaultTokenRequestSchema)],
    bank: ['bank', optional(bankRequestSchema)],
}));
