// @paypal/paypal-server-sdk/dist/esm/models/subscriptionBillingInformation.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, number, object, optional, string, } from '../schema.js';
import { cycleExecutionSchema } from './cycleExecution.js';
import { failedPaymentDetailsSchema, } from './failedPaymentDetails.js';
import { lastPaymentDetailsSchema, } from './lastPaymentDetails.js';
import { moneySchema } from './money.js';
export const subscriptionBillingInformationSchema = lazy(() => object({
    outstandingBalance: ['outstanding_balance', moneySchema],
    cycleExecutions: [
        'cycle_executions',
        optional(array(cycleExecutionSchema)),
    ],
    lastPayment: ['last_payment', optional(lastPaymentDetailsSchema)],
    nextBillingTime: ['next_billing_time', optional(string())],
    finalPaymentTime: ['final_payment_time', optional(string())],
    failedPaymentsCount: ['failed_payments_count', number()],
    lastFailedPayment: [
        'last_failed_payment',
        optional(failedPaymentDetailsSchema),
    ],
}));
