// @paypal/paypal-server-sdk/dist/esm/models/transactionDetails.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional } from '../schema.js';
import { auctionInformationSchema, } from './auctionInformation.js';
import { cartInformationSchema } from './cartInformation.js';
import { incentiveInformationSchema, } from './incentiveInformation.js';
import { payerInformationSchema, } from './payerInformation.js';
import { shippingInformationSchema, } from './shippingInformation.js';
import { storeInformationSchema, } from './storeInformation.js';
import { transactionInformationSchema, } from './transactionInformation.js';
export const transactionDetailsSchema = lazy(() => object({
    transactionInfo: [
        'transaction_info',
        optional(transactionInformationSchema),
    ],
    payerInfo: ['payer_info', optional(payerInformationSchema)],
    shippingInfo: ['shipping_info', optional(shippingInformationSchema)],
    cartInfo: ['cart_info', optional(cartInformationSchema)],
    storeInfo: ['store_info', optional(storeInformationSchema)],
    auctionInfo: ['auction_info', optional(auctionInformationSchema)],
    incentiveInfo: ['incentive_info', optional(incentiveInformationSchema)],
}));
