// @paypal/paypal-server-sdk/dist/cjs/models/paypalWallet.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paypalWalletSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var name_js_1 = require("./name.js");
var paypalWalletAttributes_js_1 = require("./paypalWalletAttributes.js");
var paypalWalletExperienceContext_js_1 = require("./paypalWalletExperienceContext.js");
var paypalWalletStoredCredential_js_1 = require("./paypalWalletStoredCredential.js");
var phoneWithType_js_1 = require("./phoneWithType.js");
var taxInfo_js_1 = require("./taxInfo.js");
exports.paypalWalletSchema = (0, schema_js_1.lazy)(function () {
    return (0, schema_js_1.object)({
        vaultId: ['vault_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        name: ['name', (0, schema_js_1.optional)(name_js_1.nameSchema)],
        phone: ['phone', (0, schema_js_1.optional)(phoneWithType_js_1.phoneWithTypeSchema)],
        birthDate: ['birth_date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        taxInfo: ['tax_info', (0, schema_js_1.optional)(taxInfo_js_1.taxInfoSchema)],
        address: ['address', (0, schema_js_1.optional)(address_js_1.addressSchema)],
        attributes: ['attributes', (0, schema_js_1.optional)(paypalWalletAttributes_js_1.paypalWalletAttributesSchema)],
        experienceContext: [
            'experience_context',
            (0, schema_js_1.optional)(paypalWalletExperienceContext_js_1.paypalWalletExperienceContextSchema),
        ],
        billingAgreementId: ['billing_agreement_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        storedCredential: [
            'stored_credential',
            (0, schema_js_1.optional)(paypalWalletStoredCredential_js_1.paypalWalletStoredCredentialSchema),
        ],
    });
});
