// @paypal/paypal-server-sdk/dist/cjs/models/paypalWalletVaultBase.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paypalWalletVaultBaseSchema = void 0;
var schema_js_1 = require("../schema.js");
var paypalPaymentTokenCustomerType_js_1 = require("./paypalPaymentTokenCustomerType.js");
var storeInVaultInstruction_js_1 = require("./storeInVaultInstruction.js");
var usagePattern_js_1 = require("./usagePattern.js");
var usageType_js_1 = require("./usageType.js");
exports.paypalWalletVaultBaseSchema = (0, schema_js_1.object)({
    storeInVault: ['store_in_vault', (0, schema_js_1.optional)(storeInVaultInstruction_js_1.storeInVaultInstructionSchema)],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    usagePattern: ['usage_pattern', (0, schema_js_1.optional)(usagePattern_js_1.usagePatternSchema)],
    usageType: ['usage_type', (0, schema_js_1.optional)(usageType_js_1.usageTypeSchema)],
    customerType: [
        'customer_type',
        (0, schema_js_1.optional)(paypalPaymentTokenCustomerType_js_1.paypalPaymentTokenCustomerTypeSchema),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_js_1.optional)((0, schema_js_1.boolean)()),
    ],
});
