// @paypal/paypal-server-sdk/dist/cjs/models/subscriberRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriberRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var name_js_1 = require("./name.js");
var phoneWithType_js_1 = require("./phoneWithType.js");
var shippingDetails_js_1 = require("./shippingDetails.js");
var subscriptionPaymentSource_js_1 = require("./subscriptionPaymentSource.js");
exports.subscriberRequestSchema = (0, schema_js_1.lazy)(function () {
    return (0, schema_js_1.object)({
        emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        payerId: ['payer_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        name: ['name', (0, schema_js_1.optional)(name_js_1.nameSchema)],
        shippingAddress: ['shipping_address', (0, schema_js_1.optional)(shippingDetails_js_1.shippingDetailsSchema)],
        paymentSource: [
            'payment_source',
            (0, schema_js_1.optional)(subscriptionPaymentSource_js_1.subscriptionPaymentSourceSchema),
        ],
        phone: ['phone', (0, schema_js_1.optional)(phoneWithType_js_1.phoneWithTypeSchema)],
    });
});
