// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionBillingInformation.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionBillingInformationSchema = void 0;
var schema_js_1 = require("../schema.js");
var cycleExecution_js_1 = require("./cycleExecution.js");
var failedPaymentDetails_js_1 = require("./failedPaymentDetails.js");
var lastPaymentDetails_js_1 = require("./lastPaymentDetails.js");
var money_js_1 = require("./money.js");
exports.subscriptionBillingInformationSchema = (0, schema_js_1.lazy)(function () {
    return (0, schema_js_1.object)({
        outstandingBalance: ['outstanding_balance', money_js_1.moneySchema],
        cycleExecutions: [
            'cycle_executions',
            (0, schema_js_1.optional)((0, schema_js_1.array)(cycleExecution_js_1.cycleExecutionSchema)),
        ],
        lastPayment: ['last_payment', (0, schema_js_1.optional)(lastPaymentDetails_js_1.lastPaymentDetailsSchema)],
        nextBillingTime: ['next_billing_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        finalPaymentTime: ['final_payment_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
        failedPaymentsCount: ['failed_payments_count', (0, schema_js_1.number)()],
        lastFailedPayment: [
            'last_failed_payment',
            (0, schema_js_1.optional)(failedPaymentDetails_js_1.failedPaymentDetailsSchema),
        ],
    });
});
