// @paypal/paypal-server-sdk/dist/cjs/models/transactionDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.transactionDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var auctionInformation_js_1 = require("./auctionInformation.js");
var cartInformation_js_1 = require("./cartInformation.js");
var incentiveInformation_js_1 = require("./incentiveInformation.js");
var payerInformation_js_1 = require("./payerInformation.js");
var shippingInformation_js_1 = require("./shippingInformation.js");
var storeInformation_js_1 = require("./storeInformation.js");
var transactionInformation_js_1 = require("./transactionInformation.js");
exports.transactionDetailsSchema = (0, schema_js_1.lazy)(function () {
    return (0, schema_js_1.object)({
        transactionInfo: [
            'transaction_info',
            (0, schema_js_1.optional)(transactionInformation_js_1.transactionInformationSchema),
        ],
        payerInfo: ['payer_info', (0, schema_js_1.optional)(payerInformation_js_1.payerInformationSchema)],
        shippingInfo: ['shipping_info', (0, schema_js_1.optional)(shippingInformation_js_1.shippingInformationSchema)],
        cartInfo: ['cart_info', (0, schema_js_1.optional)(cartInformation_js_1.cartInformationSchema)],
        storeInfo: ['store_info', (0, schema_js_1.optional)(storeInformation_js_1.storeInformationSchema)],
        auctionInfo: ['auction_info', (0, schema_js_1.optional)(auctionInformation_js_1.auctionInformationSchema)],
        incentiveInfo: ['incentive_info', (0, schema_js_1.optional)(incentiveInformation_js_1.incentiveInformationSchema)],
    });
});
